scala_util/src/main/scala/com/splicemachine/spark/splicemachine/ShuffledDependency.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2012 - 2018 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.splicemachine.spark.splicemachine
+
+import org.apache.spark.NarrowDependency
+import org.apache.spark.rdd.RDD
+
+class ShuffledDependency[T](rdd: RDD[T]) extends NarrowDependency[T](rdd) {
+  val partitions = util.Random.shuffle[Int, IndexedSeq]((0 to (rdd.getNumPartitions - 1)).toArray)
```

Comment:
This is a very useful enhancement.  If the shuffle is random though, it might make benchmarking difficult as performance results might differ from run to run.  Is there something else we can do besides a random shuffle to make it deterministic?  Maybe spread the partitions out evenly?  For example, if range is 0 to 99, create the list as 0,9,19,29,39,49,59,69,79,89,99,1,10,20,30 ...  Could maybe use the square root of the number of partitions as the increment value.  That's just one idea, there are probably other better ways to spread out the partitions deterministically.

Reply:
I think that's a fair requirement, however currently we are already shuffling partitions in a random order when running IMPORT. I'll create a JIRA to come up with some standardized way to get more determinism. Maybe by introducing a configuration property that always sets the seed to a constant and is only set during benchmarking.